pkg/boot/linux/load_linux_arm64.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package linux
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/boot/image"
+	"github.com/u-root/u-root/pkg/boot/kexec"
+	"github.com/u-root/u-root/pkg/dt"
+	"github.com/u-root/u-root/pkg/uio"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	kernelAlignSize = 1 << 21 // 2 MB.
+)
+
+func mmap(f *os.File) (data []byte, ummap func() error, err error) {
+	fd := f.Fd()
```

Comment:
one last nit: you declare fd but don't use it but one place. This early declaration of variables that are not used until later has been a source of bugs -- I know, I've both seen and created them. Could you consider either
1. moving this to line 34
2. just don't create the variable, you only use it in one place, and int(f.Fd()) is just as clear.

Reply:
The problem comes in if someone used the variable fd, instead of some other int variable they might use, because of a typo or similar problem. Suppose in future somebody creates an int variable called df. If they incorrectly use fd instead of df, because both are declared, the bug can be pretty unpleasant to find. I've seen that happen. 

Whereas, if the variable fd is not declared until the point of first use, they will get a compiler error if they type fd instead of df. 

It's why they advise you to declare new variables near the site where they are first used, or try to avoid declaring a new variable if you only use it once.

Reply:
I see, thanks ^ ^